test_replica_update fails in `prod`

https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/55184

I reverted 82110e81 on the `prod` branch for the time being. 

```
======================================================================
ERROR: test_replica_update (indexer.test_indexer.TestDCP1IndexerWithIndexesSetUp.test_replica_update) [New replica]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 539, in one
    first_value = next(it)
                  ^^^^^^^^
StopIteration
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/indexer/test_indexer.py", line 2133, in test_replica_update
    hit = one(self._get_all_hits())
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 541, in one
    raise (
ValueError: too few items in iterable (expected 1)
======================================================================
ERROR: test_replica_update (indexer.test_indexer.TestDCP1IndexerWithIndexesSetUp.test_replica_update) [Additional hub IDs]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 539, in one
    first_value = next(it)
                  ^^^^^^^^
StopIteration
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/indexer/test_indexer.py", line 2133, in test_replica_update
    hit = one(self._get_all_hits())
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 541, in one
    raise (
ValueError: too few items in iterable (expected 1)
======================================================================
ERROR: test_replica_update (indexer.test_indexer.TestDCP1IndexerWithIndexesSetUp.test_replica_update) [Redundant hub IDs]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 539, in one
    first_value = next(it)
                  ^^^^^^^^
StopIteration
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/indexer/test_indexer.py", line 2133, in test_replica_update
    hit = one(self._get_all_hits())
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/build/.venv/lib/python3.11/site-packages/more_itertools/more.py", line 541, in one
    raise (
ValueError: too few items in iterable (expected 1)
----------------------------------------------------------------------
Ran 419 tests in 2810.963s
FAILED (errors=3, skipped=3)
make: *** [Makefile:231: test] Error 1
Cleaning up project directory and file based variables 00:01
ERROR: Job failed: exit code 1
```

--- a/test/azul_test_case.py
+++ b/test/azul_test_case.py
@@ -359,6 +359,7 @@ class CatalogTestCase(AzulUnitTestCase, metaclass=ABCMeta):
     def setUpClass(cls) -> None:
         super().setUpClass()
         cls._patch_catalogs()
+        cls._patch_replicas_enabled()
 
     @classmethod
     def _patch_catalogs(cls):
@@ -387,6 +388,12 @@ class CatalogTestCase(AzulUnitTestCase, metaclass=ABCMeta):
         # Ensure that derived cached properties are affected
         assert config.default_catalog == cls.catalog
         assert config.integration_test_catalogs == {}
+
+    @classmethod
+    def _patch_replicas_enabled(cls):
+        cls.addClassPatch(patch.object(type(config),
+                                       'enable_replicas',
+                                       return_value=True))
 
 
 class DSSTestCase(CatalogTestCase, metaclass=ABCMeta):

--- a/test/indexer/test_indexer.py
+++ b/test/indexer/test_indexer.py
@@ -207,60 +207,66 @@ class TestDCP1Indexer(DCP1IndexerTestCase):
         for max_partition_size in [BundlePartition.max_partition_size, 1]:
             for page_size in (config.contribution_page_size, 1):
                 for enable_replicas in True, False:
-                    with self.subTest(page_size=page_size,
-                                      max_partition_size=max_partition_size,
-                                      enable_replicas=enable_replicas):
-                        with patch.object(target=type(config),
-                                          attribute='enable_replicas',
-                                          new_callable=PropertyMock,
-                                          return_value=enable_replicas):
-                            with patch.object(BundlePartition, 'max_partition_size', new=max_partition_size):
-                                with patch.object(type(config), 'contribution_page_size', new=page_size):
-                                    self.index_service.create_indices(self.catalog)
-                                    try:
-                                        self._index_bundle(bundle, delete=False)
-                                        hits = self._get_all_hits()
-                                        if enable_replicas:
-                                            expected_hits = canned_hits
-                                        else:
-                                            expected_hits = [
-                                                h
-                                                for h in canned_hits
-                                                if self._parse_index_name(h)[1] is not DocumentType.replica
-                                            ]
-                                        self.assertElasticEqual(expected_hits, hits)
-                                        contributions = []
-                                        replicas = []
-                                        for hit in hits:
-                                            qualifier, doc_type = self._parse_index_name(hit)
-                                            if doc_type is DocumentType.replica:
-                                                entity_id = ReplicaCoordinates.from_hit(hit).entity.entity_id
-                                                replicas.append(entity_id)
-                                                if entity_id == bundle.uuid:
-                                                    expected = bundle.metadata_files['links.json']
-                                                else:
-                                                    expected = one(
-                                                        m
-                                                        for m in bundle.metadata_files.values()
-                                                        if (
-                                                            m['schema_type'] != 'link_bundle'
-                                                            and m['provenance']['document_id'] == entity_id
-                                                        )
-                                                    )
-                                                # Replica contents should match the entity
-                                                # metadata as supplied by the repository
-                                                # plugin, verbatim
-                                                actual = hit['_source']['contents']
-                                                self.assertEqual(expected, actual)
-                                            elif doc_type is DocumentType.contribution:
-                                                entity_id = ContributionCoordinates.from_hit(hit).entity.entity_id
-                                                contributions.append(entity_id)
-                                        contributions.sort()
-                                        replicas.sort()
-                                        # Every contribution should be replicated
-                                        self.assertEqual(contributions if enable_replicas else [], replicas)
-                                    finally:
-                                        self.index_service.delete_indices(self.catalog)
+                    with (
+                        self.subTest(page_size=page_size,
+                                     max_partition_size=max_partition_size,
+                                     enable_replicas=enable_replicas),
+                        patch.object(target=type(config),
+                                     attribute='enable_replicas',
+                                     new_callable=PropertyMock,
+                                     return_value=enable_replicas),
+                        patch.object(BundlePartition,
+                                     'max_partition_size',
+                                     new=max_partition_size),
+                        patch.object(type(config),
+                                     'contribution_page_size',
+                                     new=page_size)
+                    ):
+                        self.index_service.create_indices(self.catalog)
+                        try:
+                            self._index_bundle(bundle, delete=False)
+                            hits = self._get_all_hits()
+                            if enable_replicas:
+                                expected_hits = canned_hits
+                            else:
+                                expected_hits = [
+                                    h
+                                    for h in canned_hits
+                                    if self._parse_index_name(h)[1] is not DocumentType.replica
+                                ]
+                            self.assertElasticEqual(expected_hits, hits)
+                            contributions = []
+                            replicas = []
+                            for hit in hits:
+                                qualifier, doc_type = self._parse_index_name(hit)
+                                if doc_type is DocumentType.replica:
+                                    entity_id = ReplicaCoordinates.from_hit(hit).entity.entity_id
+                                    replicas.append(entity_id)
+                                    if entity_id == bundle.uuid:
+                                        expected = bundle.metadata_files['links.json']
+                                    else:
+                                        expected = one(
+                                            m
+                                            for m in bundle.metadata_files.values()
+                                            if (
+                                                m['schema_type'] != 'link_bundle'
+                                                and m['provenance']['document_id'] == entity_id
+                                            )
+                                        )
+                                    # Replica contents should match the entity
+                                    # metadata as supplied by the repository
+                                    # plugin, verbatim
+                                    actual = hit['_source']['contents']
+                                    self.assertEqual(expected, actual)
+                                elif doc_type is DocumentType.contribution:
+                                    entity_id = ContributionCoordinates.from_hit(hit).entity.entity_id
+                                    contributions.append(entity_id)
+                            contributions.sort()
+                            replicas.sort()
+                            # Every contribution should be replicated
+                            self.assertEqual(contributions if enable_replicas else [], replicas)
+                        finally:
+                            self.index_service.delete_indices(self.catalog)
 
     def test_deletion(self):
         """

--- a/test/service/test_manifest.py
+++ b/test/service/test_manifest.py
@@ -1317,8 +1317,6 @@ class TestManifests(DCP1ManifestTestCase, PFBTestCase):
                         actual_cd = query.params['response-content-disposition']
                         self.assertEqual(expected_cd, actual_cd)
 
-    @unittest.skipIf(not config.enable_replicas,
-                     'The format is replica-based')
     @manifest_test
     def test_verbatim_jsonl_manifest(self):
         expected = [
@@ -2059,8 +2057,6 @@ class TestAnvilManifests(AnvilManifestTestCase):
         ]
         self._assert_tsv(expected, response)
 
-    @unittest.skipIf(not config.enable_replicas,
-                     'The format is replica-based')
     @manifest_test
     def test_verbatim_jsonl_manifest(self):
         response = self._get_manifest(ManifestFormat.verbatim_jsonl, filters={})
@@ -2075,8 +2071,6 @@ class TestAnvilManifests(AnvilManifestTestCase):
         ]
         self._assert_jsonl(expected, response)
 
-    @unittest.skipIf(not config.enable_replicas,
-                     'The format is replica-based')
     @manifest_test
     def test_verbatim_pfb_manifest(self):
         response = self._get_manifest(ManifestFormat.verbatim_pfb, filters={})

